week10/crud/src/App.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from 'react';
+import { Routes, Route, NavLink, useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import RegisterPage from './pages/Register';
+import WritePage from './pages/Write';
+import ListPage from './pages/List';
+import DetailPage from './pages/Detail';
+import HeartPage from './pages/Heart';
+
+function App() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // 로그인 여부 확인
+    const token = localStorage.getItem('efubtoken');
+    if (token) {
+      setIsLoggedIn(true);
+    } else {
+      setIsLoggedIn(false);
+    }
+  }, []);
+
+  const handleLogout = () => {
+    // 로그아웃 처리
+    localStorage.removeItem('efubtoken');
+    setIsLoggedIn(false);
+    alert("로그아웃 되었습니다.")
+    // 로그아웃 후 회원가입 페이지로
+    navigate('/register');
+  };
+
+  const renderRoutes = () => {
+    if (isLoggedIn) {
+      return (
+        <Routes>
+          <Route path="/" element={<ListPage />} />
+          <Route path="/detail/:postId" element={<DetailPage />} />
+          <Route path="/write" element={<WritePage />} />
+          <Route path="/heart" element={<HeartPage />} />
+          <Route path="*" element={<NoMatch />} />
+        </Routes>
+      );
+    } else {
+      return (
+        <Routes>
+          <Route path="/register" element={<RegisterPage />} />
+          <Route path="*" element={<navigate to="/register" />} />
+        </Routes>
+      );
+    }
+  };
+
+  return (
+    <Container>
+      <Navbar>
+        <StyledLink to="/" activeClassName="active">글 목록</StyledLink>
+        {isLoggedIn && <StyledLink to="/write" activeClassName="active">글쓰기</StyledLink>}
+        <StyledLink to="/register" activeClassName="active" onClick={handleLogout}>
+          {isLoggedIn ? '로그아웃' : '로그인/회원가입'}
+        </StyledLink>
+        {isLoggedIn && <StyledLink to="/heart" activeClassName="active">좋아요 누른 글</StyledLink>}
+      </Navbar>
+      {renderRoutes()}
+    </Container>
+  );
+}
```

Comment:
- [x] 2-1. App.js의 activeClassName -> className 수정
```suggestion
  return (
    <Container>
      <Navbar>
        <StyledLink to="/" className={({ isActive }) => (isActive ? "active" : "")}>글 목록</StyledLink>
        {isLoggedIn && <StyledLink to="/write" className={({ isActive }) => (isActive ? "active" : "")}>글쓰기</StyledLink>}
        <StyledLink to="/register" className={({ isActive }) => (isActive ? "active" : "")} onClick={handleLogout}>
          {isLoggedIn ? '로그아웃' : '로그인/회원가입'}
        </StyledLink>
        {isLoggedIn && <StyledLink to="/heart" className={({ isActive }) => (isActive ? "active" : "")}>좋아요 누른 글</StyledLink>}
      </Navbar>
      {renderRoutes()}
    </Container>
  );
}
```